### PR TITLE
[ZIP2-60] Provide FX rate endpoint & admin interface

### PR DESCRIPTION
**Pull request by kaorism**
https://zipmex.atlassian.net/jira/software/projects/ZIP2/boards/4?selectedIssue=ZIP2-60

### CODE REVIEW

File: lib/exchange_admin/fx_rates.ex [hunk 1]
```diff
@@ -0,0 +1,46 @@
+defmodule ExchangeAdmin.FxRates do
```
Comment (ukrit-himakoon) [inline lib/exchange_admin/fx_rates.ex:1]: should this one under  `lib/exchange_admin/fx_rates/fx_rates.ex`?

File: config/config.exs [hunk 2]
```diff
@@ -9,6 +9,7 @@ use Mix.Config
 
 config :exchange_admin,
   dopa_api_base_url: "https://idcard.bora.dopa.go.th/CheckCardStatus/CheckCardService.asmx",
+  exchange_api_base_url: "http://localhost:5555",
```
Comment (ukrit-himakoon) [inline config/config.exs:12]: maybe need to refactor this to separate config between `prod` and` dev` env

Comment (kaorism) [reply config/config.exs:12]: Yes, need to merge deployment branch into this.

Comment (kaorism) [reply config/config.exs:12]: ok, I will merge `deployment` into branch 😃

File: lib/exchange_admin_web/controllers/fx_rate_controller.ex [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        |> put_flash(:error, "Internal server error")
+        |> render("new.html", changeset: changeset)
+    end
+  end
+
+  def create(conn, %{"fx_rate" => fx_rate_params}) do
+    case FxRates.create_fx_rate(fx_rate_params) do
+      :ok ->
+        conn
+        |> put_flash(:info, "Fx rate updated successfully.")
```
Comment (ukrit-himakoon) [inline lib/exchange_admin_web/controllers/fx_rate_controller.ex:23]: `fx rate` or `Fx rate`. need should use the same conventional

File: lib/exchange_admin/fx_rates/fx_rate.ex [hunk 4]
```diff
@@ -0,0 +1,57 @@
+defmodule ExchangeAdmin.FxRates.FxRate do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  @primary_key false
+  embedded_schema do
+    field(:aud, :string)
```
Comment (ukrit-himakoon) [inline lib/exchange_admin/fx_rates/fx_rate.ex:7]: `:decimal`

File: lib/exchange_admin_web/templates/fx_rate/form.html.eex [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<%= form_for @changeset, @action, fn f -> %>
+  <%= if @changeset.action do %>
+    <div class="alert alert-danger">
+      <p>Oops, something went wrong! Please check the errors below.</p>
+    </div>
+  <% end %>
+
+  <%= label f, :sgd %>
+  <%= text_input f, :sgd, maxlength: 20 %>
```
Comment (ukrit-himakoon) [inline lib/exchange_admin_web/templates/fx_rate/form.html.eex:9]: `20%>`

File: test/exchange_admin/fx_rates/fx_rate_api_test.exs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    test "returns :ok when successfully create fx rates" do
+      use_cassette "fx_rate_api_create_successfully",
+        match_requests_on: [:request_body, :query] do
+        fx_rate = build(:fx_rate)
+
+        assert :ok = FxRateApi.create(fx_rate)
+      end
+    end
+
+    test "returns :error when  failed to create fx rates" do
```
Comment (ukrit-himakoon) [inline test/exchange_admin/fx_rates/fx_rate_api_test.exs:46]: `:error when  failed` to `:error when failed`

File: lib/exchange_admin_web/templates/fx_rate/form.html.eex [hunk 7]
```diff
@@ -0,0 +1,28 @@
+<%= form_for @changeset, @action, fn f -> %>
+  <%= if @changeset.action do %>
+    <div class="alert alert-danger">
+      <p>Oops, something went wrong! Please check the errors below.</p>
+    </div>
+  <% end %>
+
+  <%= label f, :aud, "AUD" %>
+  <%= text_input f, :aud , maxlength: 20%>
```
Comment (ukrit-himakoon) [inline lib/exchange_admin_web/templates/fx_rate/form.html.eex:9]: or should we use this one as `number_input`

Comment (kaorism) [reply lib/exchange_admin_web/templates/fx_rate/form.html.eex]: awesome

**Pull request merged**